base/range.jl

```diff
@@ -415,6 +415,13 @@ end
 
 function getindex{T}(v::Range{T}, i::Integer)
     @_inline_meta
+    ret = convert(T, first(v) + i - 1)
+    @boundscheck ((i > 0) & (ret <= v.stop)) || throw_boundserror(v, i)
```

Comment:
I think there's a good case to be made for `a < b < c` lowering to `(a < b) & (b < c)` instead.